design/src/rfcs/rfc0031_implementing_adaptive_retry_behavior.md

```diff
@@ -0,0 +1,162 @@
+- **Transient Error**: An error that should be retried as soon as is convenient. [Currently][transient-error-classifier], the Rust SDK treats the following as transient errors:
+    - All `500`, `502`, `503`, and `504` errors that are not **Throttling Errors**.
+    - Request dispatches that fail for timeout or IO reasons.
+    - All other timeout errors.
+    - Errors modeled with the `@retryable` or `@retryable(throttling: false)` trait.
+
+## The User Experience
+
+### Today
+
+The SDK supports the **SRB** only. Users may configure retries in the environment, AWS profile file, or programmatically. Configurable options include:
+- The maximum number of attempts to make when sending a request, including the initial attempt.
+- The initial multiplier to use when calculating an exponential backoff.
+- The retry "mode". The only valid value is `standard`.
+```rust
+use std::time::Duration;
+use aws_smithy_types::retry::RetryConfigBuilder;
+// Create a retry config. This example sets them all for demonstration purposes.
+let retry_config = RetryConfigBuilder::standard()
+    .max_attempts(3)
+    .initial_backoff(Duration::from_secs(1))
+    .build();
+
+let sdk_config = aws_config::from_env()
+    .retry_config(retry_config)
+    .load()
+    .await;
+```
+
+### Tomorrow
+
+Once implemented, users can opt in to **ARB** in much the same way:
+
+```rust
+use std::time::Duration;
+use aws_smithy_types::retry::RetryConfigBuilder;
+// Create a retry config. This example sets them all for demonstration purposes.
+let retry_config = RetryConfigBuilder::adaptive()
+    .max_attempts(3)
+    .initial_backoff(Duration::from_secs(1))
+    .build();
+
+let sdk_config = aws_config::from_env()
+    .retry_config(retry_config)
+    .load()
+    .await;
+```
+
+Because **ARB** works alongside **SRB** and isn't configurable, no new config fields will be added.
+
+## The Implementation
+
+### Configuration
+
+The configuration changes required are very small as the new mode is not configurable. We already allow users to set the retry mode; We need only update it to allow users to set `adaptive` as well.
+
+```rust
+const VALID_RETRY_MODES: &[RetryMode] = &[
+    RetryMode::Standard,
+    RetryMode::Adaptive,
+];
+
+impl FromStr for RetryMode {
+    type Err = RetryModeParseError;
+
+    fn from_str(string: &str) -> Result<Self, Self::Err> {
+        let string = string.trim();
+        // eq_ignore_ascii_case is OK here because the only strings we need to check for are ASCII
+        if string.eq_ignore_ascii_case("standard") {
+            Ok(RetryMode::Standard)
+        } else if string.eq_ignore_ascii_case("adaptive") {
+            Ok(RetryMode::Adaptive)
+        } else {
+            Err(RetryModeParseError::new(string))
+        }
+    }
+}
+```
+
+### The Service
+
+Because `tower::Service`s already define an async `ready` method, we'll use that to block sending requests unless they can acquire a token from the token bucket.
```

Comment:
I am assuming that in the orchestrator world, we are going to replace this section in the RFC?